python/rascaline/rascaline/utils/clebsch_gordan/_density_correlations.py

```diff
@@ -0,0 +1,226 @@
+
+All of these calculators wrap the :py:class:`TensorCorrelator` class, handling the
+higher-level metadata manipulation to produce the desired output tensors.
+"""
+
+from typing import List, Optional
+
+from .. import _dispatch
+from .._backend import Labels, TensorMap, TorchModule, torch_jit_export
+from ._tensor_correlator import TensorCorrelator
+from ._utils import index_property_names
+
+
+class DensityCorrelations(TorchModule):
+    """
+    Takes ``n_correlations`` of iterative CG tensor products of a density with itself to
+    produce a density auto-correlation tensor of higher correlation order.
+
+    The constructor computes and stores the CG coefficients. The :py:meth:`compute`
+    method computes the auto-coorelation by CG tensor product of the input density.
+
+    :param n_correlations: :py:class:`int`, the number of iterative CG tensor products
+        to perform.
+    :param max_angular: :py:class:`int`, the maximum angular momentum to compute CG
+        coefficients for. This must be large enough to perform the desired number of
+        correlations on the denisty passed to the :py:meth:`compute` method.
+    :param arrays_backend: :py:class:`str`, the backend to use for array operations. If
+        ``None``, the backend is automatically selected based on the environment.
+        Possible values are "numpy" and "torch". Used to initialize a new
+        :py:class:`TensorCorrelator`, and so should only be passed if
+        ``tensor_correlator`` is ``None``.
+    :param cg_backend: :py:class:`str`, the backend to use for the CG tensor product. If
+        ``None``, the backend is automatically selected based on the arrays backend.
+        Used to initialize a new :py:class:`TensorCorrelator`, and so should only be
+        passed if ``tensor_correlator`` is ``None``.
+    """
+
+    def __init__(
+        self,
+        n_correlations: int,
+        max_angular: int,
+        arrays_backend: Optional[str] = None,
+        cg_backend: Optional[str] = None,
+    ) -> None:
+
+        super().__init__()
+
+        self._n_correlations = n_correlations
+        self._tensor_correlator = TensorCorrelator(
+            max_angular=max_angular,
+            arrays_backend=arrays_backend,
+            cg_backend=cg_backend,
+        )
+
+    def forward(
+        self,
+        density: TensorMap,
+        angular_cutoff: Optional[int] = None,
+        selected_keys: Optional[Labels] = None,
+        skip_redundant: bool = True,
+    ) -> TensorMap:
+        """
+        Calls the :py:meth:`compute` method.
+
+        This is intended for :py:class:`torch.nn.Module` compatibility, and should be
+        ignored in pure Python mode.
+
+        See :py:meth:`compute` for a full description of the parameters.
+        """
+        return self.compute(
+            density, selected_keys=selected_keys, skip_redundant=skip_redundant
+        )
+
+    @torch_jit_export
```

Comment:
why the export?

Reply:
I seem to remember the torch script compiler not liking it when I removed it, but will check again